tests/rptest/services/keycloak.py

```diff
@@ -28,24 +40,61 @@
 LAUNCH_JBOSS_IN_BACKGROUND=1 \
 KEYCLOAK_ADMIN={admin} \
 KEYCLOAK_ADMIN_PASSWORD={pw} \
-{kc} start-dev --http-port={port} --hostname={host} --hostname-port={port} --http-enabled=true --proxy=passthrough \
---log="{log_handler}" --log-file="{logfile}" --log-level="{log_level}" &
+{kc} start-dev
 """
 
 OIDC_CONFIG_TMPL = """\
-http://{host}:{port}/realms/{realm}/.well-known/openid-configuration\
+{scheme}://{host}:{port}/realms/{realm}/.well-known/openid-configuration\
 """
 
+DEFAULT_CONFIG = {
+    'http-port': KC_PORT,
+    'hostname': None,
+    'hostname-port': KC_PORT,
+    'http-enabled': True,
+    'proxy': 'passthrough',
+    'log': KC_LOG_HANDLER,
+    'log-file': KC_LOG_FILE,
+    'log-level': KC_ROOT_LOG_LEVEL,
+}
+
+
+class KeycloakConfigWriter:
+    _cfg = DEFAULT_CONFIG.copy()
+
+    def __init__(self, dest_dir: str = KC_CONF_DIR, extra_cfg: dict = {}):
+        self._cfg.update(extra_cfg)
+        self._dest_dir = dest_dir
+
+    @property
+    def rep(self):
+        return json.dumps(self._cfg)
+
+    def write(self, node):
+        try:
+            node.account.mkdirs(self._dest_dir)
+        except:
+            pass
+        with tempfile.NamedTemporaryFile(delete=False, mode='w') as f:
```

Comment:
I think you want `delete_on_close=False`. Setting `delete=False` I think will disable GC of the temporary file so the temporary files will be accumulated.

Reply:
> Changed in version 3.12: Added delete_on_close parameter.

Oh my... it's very new feature. Probably too new for our docker containers to have it. That's ok! You can fix it like this:

```python
with tempfile.NamedTemporaryFile() as f:
    f.write(...)
    f.flush() # make visible, but leave open
    node.account.copy_to(f.name, ...)
```

Reply:
>Setting delete=False I think will disable GC of the temporary file

Huh, I must have misread the docstring that popped up in my editor for 2s 🙃

Reply:
Gonna let this build ride so I can get my CDT run, but will fix locally

Reply:
fixed